### ISSUE TITLE
Python lists sometimes get interpreted as regex in completer

### ISSUE THREAD

**Issue by Danmou** [xonsh/xonsh#2534]
I've found a weird bug: If I write `[1-0.1]` in xonsh, I will get a traceback like the following upon writing the last square bracket:

```
Exception in thread Thread-2577:
Traceback (most recent call last):                                                                        
  File "/home/daniel/anaconda3/lib/python3.6/threading.py", line 916, in _bootstrap_inner                
    self.run()                                                                                           
  File "/home/daniel/anaconda3/lib/python3.6/threading.py", line 864, in run                             
    self._target(*self._args, **self._kwargs)                                                            
  File "/home/daniel/anaconda3/lib/python3.6/site-packages/prompt_toolkit/interface.py", line 840, in run 
    completions = list(buffer.completer.get_completions(document, complete_event))
  File "/home/daniel/Git/xonsh/xonsh/ptk/completer.py", line 49, in get_completions
    self.ctx)
  File "/home/daniel/Git/xonsh/xonsh/completer.py", line 35, in complete
    out = func(prefix, line, begidx, endidx, ctx)
  File "/home/daniel/Git/xonsh/xonsh/completers/path.py", line 268, in complete_path
    sort_result=glob_sorted):
  File "/home/daniel/Git/xonsh/xonsh/tools.py", line 1872, in iglobpath
    return _iglobpath(s, ignore_case=ignore_case, sort_result=sort_result)[0]
  File "/home/daniel/Git/xonsh/xonsh/tools.py", line 1854, in _iglobpath
    paths = glob.glob(s, recursive=True)
  File "/home/daniel/anaconda3/lib/python3.6/glob.py", line 20, in glob
    return list(iglob(pathname, recursive=recursive))
  File "/home/daniel/anaconda3/lib/python3.6/glob.py", line 55, in _iglob
    yield from _glob1(dirname, basename, dironly)
  File "/home/daniel/anaconda3/lib/python3.6/glob.py", line 83, in _glob1
    return fnmatch.filter(names, pattern)
  File "/home/daniel/anaconda3/lib/python3.6/fnmatch.py", line 52, in filter
    match = _compile_pattern(pat)
  File "/home/daniel/anaconda3/lib/python3.6/fnmatch.py", line 46, in _compile_pattern
    return re.compile(res).match
  File "/home/daniel/anaconda3/lib/python3.6/re.py", line 233, in compile
    return _compile(pattern, flags)
  File "/home/daniel/anaconda3/lib/python3.6/re.py", line 301, in _compile
    p = sre_compile.compile(pattern, flags)
  File "/home/daniel/anaconda3/lib/python3.6/sre_compile.py", line 562, in compile
    p = sre_parse.parse(p, flags)
  File "/home/daniel/anaconda3/lib/python3.6/sre_parse.py", line 856, in parse
    p = _parse_sub(source, pattern, flags & SRE_FLAG_VERBOSE, False)
  File "/home/daniel/anaconda3/lib/python3.6/sre_parse.py", line 415, in _parse_sub
    itemsappend(_parse(source, state, verbose))
  File "/home/daniel/anaconda3/lib/python3.6/sre_parse.py", line 763, in _parse
    p = _parse_sub(source, state, sub_verbose)
  File "/home/daniel/anaconda3/lib/python3.6/sre_parse.py", line 415, in _parse_sub
    itemsappend(_parse(source, state, verbose))
  File "/home/daniel/anaconda3/lib/python3.6/sre_parse.py", line 552, in _parse
    raise source.error(msg, len(this) + 1 + len(that))
sre_constants.error: bad character range 1-0 at position 5
```

It seems like the completer module thinks I'm writing a regex glob pattern and tries to parse it.

**Comment by Danmou** [comment]
As a side note, this is another problem that falls through the exception handling as discussed in #2519.

**Comment by gforsyth** [comment]
Hey @Danmou, this was fixed in #2535 and should be in the recent 0.6.0 release
